src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineComputerLauncher.java

```diff
@@ -305,7 +326,16 @@ private void launch(ComputeEngineComputer computer, TaskListener listener) {
       conn = cleanupConn.get();
       String javaExecPath = node.getJavaExecPathOrDefault();
       if (!checkJavaInstalled(computer, conn, logger, listener, javaExecPath)) {
-        return;
+        if (ENABLE_JAVA_INSTALLATION) {
+          logInfo(computer, listener, "Let's install java for some *nix flavours");
+          if (!installJava(computer, conn, logger, listener, "sudo yum install -y java-1.8.0-openjdk.x86_64")) {
```

Comment:
I wonder if there might not be a Chocolatey/PS analog for Windows we could employ.

Reply:
I think so, let's see how it goes with the upcoming iterations first. I'll include a task in this PR to see whether we can support windows too.